src/corecel/cont/Bitset.hh

```diff
@@ -0,0 +1,310 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/cont/Bitset.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <climits>
+#include <cstddef>
+
+#include "corecel/Config.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/cont/detail/BitsetUtils.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Device-compatible bitset implementation.
+ *
+ * This implementation is based on libstdc++'s std::bitset implementation.
+ * It it a subset of the C++ standard, but it should be sufficient
+ * for our current use case. Given that GPU typically use 32-bit words, this
+ * uses unsigned int as the word type instead of the unsigned long used by the
+ * standard library.
+ *
+ * The following methods are not implemented:
+ * - conversions to string, to_ulong, to_ullong
+ * - set operations with other bitsets
+ * - stream operators
+ * - hash support
+ * - construct from string, from_ulong, from_ullong
+ */
+template<size_t N>
```

Comment:
Also, the libstdc++ uses a base class to specialize in case we only need a single word, we might want to do that as well since it seems like this would be the most common use case?

Reply:
Before we go through the trouble, maybe we could compile our "unoptimized" and the libstdc++ "optimized" on godbolt and see if there's any actual differences. Since the size is known at compile time and everything's constexpr, I *hope* the compiler could make them equivalent?

Reply:
gcc and clang seem to optimize that: https://godbolt.org/z/Eqjvj17PK

Reply:
Yep, that should be good for now!